### PR TITLE
Zms 86 mailboxes

### PR DESCRIPTION
**Pull request by NickOvt**
- [x] tests for POST `/users/{user}/mailboxes`
- [x] tests for GET `/users/{user}/mailboxes` 
- [x] tests for GET `/users/:user/mailboxes/:mailbox`
- [x] tests for PUT `/users/:user/mailboxes/:mailbox`
- [x] tests for DELETE  `/users/:user/mailboxes/:mailbox`

### CODE REVIEW

File: test/api/mailboxes-test.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+const supertest = require('supertest');
+const chai = require('chai');
+
+const expect = chai.expect;
+chai.config.includeStack = true;
+const config = require('wild-config');
+
+const server = supertest.agent(`http://127.0.0.1:${config.api.port}`);
+
+describe('Storage tests', function () {
```
Comment (andris9) [inline test/api/mailboxes-test.js:16]: Is this a typo? Shouldn't it say Mailboxes tests? Applies to the user generation request below as well.

File: test.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+test file new for git
```
Comment (andris9) [inline test.txt:1]: What is the reason for this file? Does not seem to be referenced in the tests 🤔

**Review by andris9 — CHANGES_REQUESTED** [review]

**Comment by andris9** [comment]
Does the `in progress` and `TODO` mean that you will be updating this PR with additional tests or is it done and you add these tests later or in a separate PR?

File: test.txt (hunk 2 above)
Comment (NickOvt) [reply test.txt:1]: For some reason git seems to pick it up sometimes from my local master branch, will investigate. Reason was to test git setup, but the file shouldn't have been pushed to this repo

**Comment by NickOvt** [comment]
Yes, will add tests for other 3 endpoints under this PR as well, just different commits.

File: test/api/mailboxes-test.js [hunk 3]
```diff
@@ -146,88 +146,315 @@ describe('Mailboxes tests', function () {
     });
 
     it('should GET /users/{user}/mailboxes expect success / all params', async () => {
-        const response = await server.get(`/users/${user}/mailboxes`).send({ specialUse: true, showHidden: true, counters: true, sizes: true }).expect(200);
+        const response = await server.get(`/users/${user}/mailboxes?specialUse=true&showHidden=true&counters=true&sizes=true`).send({}).expect(200);
```
Comment (andris9) [inline test/api/mailboxes-test.js:149]: Here and in other places - you don't need `send()` for GET requests. You can shorten it to `server.get(...).expect(200)`

**Review by andris9 — CHANGES_REQUESTED** [review]

File: test/api/mailboxes-test.js (hunk 3 above)
Comment (NickOvt) [reply test/api/mailboxes-test.js]: Fixed

**Review by andris9 — APPROVED** [review]

**Pull request merged**